### ISSUE TITLE
Document best practices on how to check permissions for inserting/creating an object

### ISSUE THREAD

**Issue by javiertury** [ory/keto#728]
I couldn't find information on which guidelines to follow when checking if a subject has permissions to create a **new object**.

It's clear that relations for **existing objects** must have this structure

```
namespace:object_id#relation@user
```

But how to manage permissions to insert/create a new object? Consider the case in which the object has a parent in another namespace. Should the permission be declared in a parent object (if it exists, or if there is only one type of parent)? What happens if a parent namespace can have multiple types of children? Should the relation then contain the name of the child namespace?

```
author:1234#insert_book@user
author:1234#insert_painting@user
author:1234#insert_article@user
```

Now consider the case where this object is in a root namespace and has no parent. Should the object_id be left blank? Should the object_id contain a special token?

```
author:???#insert@user
author:insert#insert@user
author:#insert@user
```

**Comment by zepatrik** [comment]
TL;DR
Whatever fits your needs and scales best with your data model is probably the right way to go.

Good point, and good considerations already. Did you manage to get an example up and running that works for you?
I would from a first glance say that all of you proposed solutions have advantages and disadvantages depending on the use-case. Maybe all of them are valid depending on the case?

Things to consider are:
- ensure special IDs are truly special by preventing clashes with normal IDs
- in a hierarchical structure, you probably want to tie the relation-tuple to the parent somehow
- in case of a varying or large number of operations that are not specific to an object ID, you might want to group them in an `operations` namespace

I will keep this issue for collecting approaches, and then we can make an overview docs page about this topic once we have a view possibilities.

**Comment by javiertury** [comment]
> in case of a varying or large number of operations that are not specific to an object ID, you might want to group them in an operations namespace

Your solution looks more elegant, thanks!

> Did you manage to get an example up and running that works for you?

Not yet, I'm still new to the ory ecosystem
